jena-arq/src/main/java/org/apache/jena/riot/out/NodeFormatterTTL_MultiLine.java

```diff
@@ -18,95 +18,120 @@
  */
 public class NodeFormatterTTL_MultiLine extends NodeFormatterTTL {
-    /// For '-quoted and "-quoted literals 
-    private final QuotedStringOutput escapeProc2 = new QuotedStringOutputTTL_MultiLine(CH_QUOTE2) ;
-    private final QuotedStringOutput escapeProc1 = new QuotedStringOutputTTL_MultiLine(CH_QUOTE1) ;
+    /// For '-quoted and "-quoted literals
+    private final QuotedStringOutput escapeProc2 = new QuotedStringOutputTTL_MultiLine(CH_QUOTE2);
+    private final QuotedStringOutput escapeProc1 = new QuotedStringOutputTTL_MultiLine(CH_QUOTE1);
 
     private boolean writeAsMultiLine(AWriter w, String lex) {
-        return ( lex.contains("\n") ) ;
+        return (lex.contains("\n"));
     }
 
     public NodeFormatterTTL_MultiLine(String baseIRI, PrefixMap prefixMap) {
-        super(baseIRI, prefixMap) ;
+        super(baseIRI, prefixMap);
     }
 
     public NodeFormatterTTL_MultiLine(String baseIRI, PrefixMap prefixMap, NodeToLabel nodeToLabel) {
         super(baseIRI, prefixMap, nodeToLabel);
     }
 
-    private Runnable noop = ()->{} ;
+    private Runnable noop = () -> {};
 
     @Override
-    public void formatLitString(AWriter w, String lex)
-    {
-        if ( ! writeAsMultiLine(w, lex) ) {
-            writeLexicalSingleLine(w, lex, noop);
-            return ;
+    public void formatLitString(AWriter w, String lex) {
+        if ( !writeAsMultiLine(w, lex) ) {
+            // super.formatLitString(w, lex);
+            writeLexicalSingleLine(w, lex, noop); // switching quotes as needed.
+            return;
         }
-        writeLexicalMultiLine(w, lex, noop) ; 
+        writeLexicalMultiLine(w, lex, noop);
     }
 
-    // To do ....
     @Override
     public void formatLitLang(AWriter w, String lex, String langTag) {
-        if ( ! writeAsMultiLine(w, lex) ) {
-            super.formatLitLang(w, lex, langTag);
-            return ;
+        Runnable runnable = () -> {
+            w.print('@');
+            w.print(langTag);
+        };
+
+        if ( !writeAsMultiLine(w, lex) ) {
+            // super Not OK.
+            // super.formatLitLang(w, lex, langTag);
+            writeLexicalSingleLine(w, lex, runnable);
+            return;
+        }
+        writeLexicalMultiLine(w, lex, runnable);
+    }
+
+    @Override
+    public void formatLitDT(AWriter w, String lex, String datatype) {
+        Runnable runnable = () -> {
+            w.print("^^");
+            formatURI(w, datatype);
+        };
+        if ( !writeAsMultiLine(w, lex) ) {
+            writeLexicalSingleLine(w, lex, runnable);
+            return;
         }
-        writeLexicalMultiLine(w, lex, ()->{ w.print('@') ; w.print(langTag) ; }) ;
+        writeLexicalMultiLine(w, lex, runnable);
     }
 
     @Override
-    protected void writeLiteralLongForm(AWriter w, String lex, String datatypeURI)
-    {
-        if ( ! writeAsMultiLine(w, lex) ) {
+    protected void writeLiteralLongForm(AWriter w, String lex, String datatypeURI) {
```

Comment:
This method is no longer necessary. It is called only from `NodeFormatterTTL.formatLitDT(AWriter w, String lex, String datatypeURI)` (i.e. in this class's superclass). But, since this class now overrides `formatLitDT(AWriter w, String lex, String datatypeURI)` and implements it differently, this method  will never be called.

Reply:
Overriding `formatLitDT` was wrong. It didn't pick up abbreviated forms. Test added.